node/bft/src/primary.rs

```diff
@@ -96,6 +98,8 @@ pub struct Primary<N: Network> {
     handles: Arc<Mutex<Vec<JoinHandle<()>>>>,
     /// The lock for propose_batch.
     propose_lock: Arc<TMutex<u64>>,
+    /// The outstanding requests to a peer.
+    outstanding_certificate_requests: Arc<Mutex<HashMap<(SocketAddr, u64), u32>>>,
```

Comment:
Similarly, here it would be good to write what this maps. Something like "(peer_ip x round_number) -> num_outstanding_requests" would be sufficient.

Maybe in the future we can have a round number type. u64 is always a little less descriptive than I would like.